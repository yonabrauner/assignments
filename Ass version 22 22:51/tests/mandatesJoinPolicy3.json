{
  "parties": [
    {
      "name": "a",
      "mandates": 45,
      "join_policy": "M"
    },
    {
      "name": "b",
      "mandates": 25,
      "join_policy": "L"
    },
    {
      "name": "c",
      "mandates": 25,
      "join_policy": "L"
    },
    {
      "name": "d",
      "mandates": 25,
      "join_policy": "L"
    }
  ],
  "graph": [
    [0, 1, 1, 1],
    [1, 0, 0, 0],
    [1, 0, 0, 0],
    [1, 0, 0, 0]
  ],
  "agents": [
    {
      "party_id": 1,
      "selection_policy": "E"
    },
    {
      "party_id": 2,
      "selection_policy": "E"
    },
    {
      "party_id": 3,
      "selection_policy": "E"
    }
  ]
}
