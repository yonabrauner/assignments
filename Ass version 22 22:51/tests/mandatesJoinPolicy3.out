[
    {
        "agents": [
            {
                "id": 0,
                "partyId": 1
            },
            {
                "id": 1,
                "partyId": 2
            },
            {
                "id": 2,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 25,
                "parties": [
                    1
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 45,
                "name": "a",
                "state": "Waiting"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "b",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "c",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 25,
                "name": "d",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 1
            },
            {
                "id": 1,
                "partyId": 2
            },
            {
                "id": 2,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 25,
                "parties": [
                    1
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 45,
                "name": "a",
                "state": "CollectingOffers"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "b",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "c",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 25,
                "name": "d",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 1
            },
            {
                "id": 1,
                "partyId": 2
            },
            {
                "id": 2,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 25,
                "parties": [
                    1
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 45,
                "name": "a",
                "state": "CollectingOffers"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "b",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "c",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 25,
                "name": "d",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 1
            },
            {
                "id": 1,
                "partyId": 2
            },
            {
                "id": 2,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 25,
                "parties": [
                    1
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 45,
                "name": "a",
                "state": "CollectingOffers"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "b",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "c",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 25,
                "name": "d",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 1
            },
            {
                "id": 1,
                "partyId": 2
            },
            {
                "id": 2,
                "partyId": 3
            },
            {
                "id": 3,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 70,
                "parties": [
                    1,
                    0
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    2
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 45,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "b",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "c",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 25,
                "name": "d",
                "state": "Joined"
            }
        ]
    }
]
